### ISSUE TITLE
false positive E1137 on pandas dataframe column assignment when set_index() is used

### ISSUE THREAD

**Issue by jmrouet** [PyCQA/pylint#7564]
### Bug description

# pylint: disable=missing-module-docstring,invalid-name
import pandas as pd

df = pd.DataFrame([{'id': 0, 'name': 'test'}]).set_index('id')
df['newcolumn'] = 'foo'
print(df)

### Command used

```shell
pylint test2.py
```

### Pylint output

```shell
************* Module test2
test2.py:5:0: E1137: 'df' does not support item assignment (unsupported-assignment-operation)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 0.00/10, +10.00)

### Pylint version

```shell
pylint 2.15.2
astroid 2.12.9
Python 3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:51:29) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 11

### Additional dependencies

pandas==1.5.0

**Comment by jacobtylerwalls** [comment]
This bug is one of the several reported in the bug cluster #4577. May be worth keeping this issue in order to break them out.

**Comment by wshanks** [comment]
I am not sure if it makes sense to treat this case separately from #4577, but I just wanted to note that this particular case only started showing up for me with pandas 1.5.0 while the other cases in #4577 were present for earlier versions of pandas.

**Comment by jmrouet** [comment]
Just to update a little bit on this matter, I tested this morning with more recent package versions:
```
astroid                           2.15.6
pandas                            2.1.0
pylint                            2.17.5
Python        3.9.16
```
but the issue still remains and I don't see any activity or progress on #457
